### PR TITLE
[Session] Do not use deprecated session service injections

### LINKED ISSUE #11042: [Symfony] Do not use deprecated Session service injections
### Improvement description

Please see https://github.com/pimcore/pimcore/pull/11041

https://symfony.com/blog/new-in-symfony-5-3-session-service-deprecation

### PR DESCRIPTION
**Pull request by dvesh3**
## Changes in this pull request  
Resolves #11042

## Additional info  
- [x] Registering a session bag through Session Configurator will not work anymore on Symfony 6 as Session service is removed. Use Event Listeners to register session bags before the session is started. Done here https://github.com/pimcore/pimcore/pull/12294

### CODE REVIEW

File: bundles/AdminBundle/Session/Handler/AdminSessionHandler.php [hunk 1]
```diff
@@ -69,22 +65,22 @@ public function __construct(SessionInterface $session, RequestHelper $requestHel
      */
     public function getSessionId()
     {
-        if (!$this->session->isStarted()) {
+        if ($this->requestHelper->getSession()->isStarted()) {
```
Comment (blankse) [inline bundles/AdminBundle/Session/Handler/AdminSessionHandler.php]: Missing ! sign?
```suggestion
        if (!$this->requestHelper->getSession()->isStarted()) {
```

File: bundles/CoreBundle/DependencyInjection/Compiler/SessionConfiguratorPass.php [hunk 2]
```diff
@@ -31,16 +31,18 @@ final class SessionConfiguratorPass implements CompilerPassInterface
...
     public function process(ContainerBuilder $container)
     {
-        if (!$container->hasDefinition('session')) {
+        // @phpstan-ignore-next-line
+        if (!$container->has('session')) {
             return;
         }
 
+        // @phpstan-ignore-next-line
         if (!$container->hasDefinition(SessionConfigurator::class)) {
```
Comment (fashxp) [inline bundles/CoreBundle/DependencyInjection/Compiler/SessionConfiguratorPass.php]: why not here `has` too?

File: bundles/CoreBundle/Resources/config/services.yaml [hunk 3]
```diff
@@ -6,15 +6,6 @@ services:
     #
     # SESSION
     #
-
```
Comment (fashxp) [inline bundles/CoreBundle/Resources/config/services.yaml:9]: can we just remove that service without BC break?

File: bundles/EcommerceFrameworkBundle/CartManager/SessionCart.php [hunk 4]
```diff
@@ -44,7 +44,7 @@ protected function getCartCheckoutDataClassName()
     protected static function getSessionBag(): AttributeBagInterface
     {
         /** @var AttributeBagInterface $sessionBag */
-        $sessionBag = \Pimcore::getContainer()->get('session')->getBag(SessionConfigurator::ATTRIBUTE_BAG_CART);
+        $sessionBag = \Pimcore::getContainer()->get('request_stack')->getSession()->getBag(SessionConfigurator::ATTRIBUTE_BAG_CART);
```
Comment (fashxp) [inline bundles/EcommerceFrameworkBundle/CartManager/SessionCart.php]: fallback needed in case of not available `request_stack`

File: bundles/EcommerceFrameworkBundle/PricingManager/PricingManager.php [hunk 5]
```diff
@@ -227,8 +230,9 @@ public function getValidRules()
      */
     public function getEnvironment()
     {
+        $requestStack = \Pimcore::getContainer()->get('request_stack');
```
Comment (fashxp) [inline bundles/EcommerceFrameworkBundle/PricingManager/PricingManager.php]: fallback needed in case of not available `request_stack`?

File: bundles/EcommerceFrameworkBundle/SessionEnvironment.php [hunk 6]
```diff
@@ -53,14 +62,27 @@ public function __construct(SessionInterface $session, LocaleServiceInterface $l
...
+     * @TODO move to constructor injection in Pimcore 11
+     *
+     * @required
+     * @internal
+     *
+     * @param RequestStack $requestStack
+     */
+    public function setRequestStack(RequestStack $requestStack): void
+    {
+        $this->requestStack = $requestStack;
```
Comment (fashxp) [inline bundles/EcommerceFrameworkBundle/SessionEnvironment.php:76]: what if request stack is not available in case of CLI?

Comment (fashxp) [reply bundles/EcommerceFrameworkBundle/SessionEnvironment.php:76]: or is request stack always available?

File: bundles/EcommerceFrameworkBundle/Tracking/TrackingManager.php [hunk 7]
```diff
@@ -78,6 +87,19 @@ public function setSession(SessionInterface $session)
         $this->session = $session;
```
Comment (fashxp) [inline bundles/EcommerceFrameworkBundle/Tracking/TrackingManager.php:89]: add deprecation to `setSession`

File: tests/Model/Permissions/ModelAssetPermissionsTest.php [hunk 8]
```diff
@@ -725,23 +725,23 @@ public function testManyElementSearch()
         //update role
         $role = User\Role::getByName('Testrole');
         $role->setWorkspacesAsset([
-            (new User\Workspace\Asset())->setValues(['cId' => $manyElementX->getId(), 'cPath' => $manyElementX->getFullpath(), 'list' => true, 'view' => true]),
-            (new User\Workspace\Asset())->setValues(['cId' => $this->groupfolder->getId(), 'cPath' => $this->groupfolder->getFullpath(), 'list' => true, 'view' => true]),
+            (new User\Workspace\Asset())->setValues(['cId' => $manyElementX->getId(), 'cPath' => $manyElementX->getRealFullPath(), 'list' => true, 'view' => true]),
```
Comment (fashxp) [inline tests/Model/Permissions/ModelAssetPermissionsTest.php:728]: part of that PR?

**Review by fashxp — CHANGES_REQUESTED** [review]

File: bundles/AdminBundle/Session/Handler/AdminSessionHandler.php (hunk 1 above)
Comment (fashxp) [inline bundles/AdminBundle/Session/Handler/AdminSessionHandler.php]: missing `!`

**Review by fashxp — CHANGES_REQUESTED** [review]

File: bundles/CoreBundle/DependencyInjection/Compiler/SessionConfiguratorPass.php (hunk 2 above)
Comment (dvesh3) [reply bundles/CoreBundle/DependencyInjection/Compiler/SessionConfiguratorPass.php]: changed it.

File: bundles/CoreBundle/Resources/config/services.yaml (hunk 3 above)
Comment (dvesh3) [reply bundles/CoreBundle/Resources/config/services.yaml:9]: It was an override to use `NamespacedAttributeBag` in session service from Symfony. We don't use NamespacedAttributeBag anymore please see https://github.com/pimcore/pimcore/pull/10909

File: bundles/EcommerceFrameworkBundle/CartManager/SessionCart.php (hunk 4 above)
Comment (dvesh3) [reply bundles/EcommerceFrameworkBundle/CartManager/SessionCart.php]: I don't think so there's another way to get the session on Symfony 6 other than request.

File: bundles/AdminBundle/Session/Handler/AdminSessionHandler.php (hunk 1 above)
Comment (dvesh3) [reply bundles/AdminBundle/Session/Handler/AdminSessionHandler.php]: fixed it

File: tests/Model/Permissions/ModelAssetPermissionsTest.php (hunk 8 above)
Comment (dvesh3) [reply tests/Model/Permissions/ModelAssetPermissionsTest.php:728]: Not really....this is required for the tests to pass. when using `getFullpath()` the path is generated per site for frontend request and it fails when no request is available and we just want to test the permissions in the test.

File: bundles/EcommerceFrameworkBundle/Tracking/TrackingManager.php (hunk 7 above)
Comment (dvesh3) [reply bundles/EcommerceFrameworkBundle/Tracking/TrackingManager.php:89]: done

File: bundles/EcommerceFrameworkBundle/SessionEnvironment.php (hunk 6 above)
Comment (dvesh3) [reply bundles/EcommerceFrameworkBundle/SessionEnvironment.php:76]: How to call this in CLI? AFAIK this is used in the Guest User Cart initialization.

File: bundles/CoreBundle/DependencyInjection/Compiler/SessionConfiguratorPass.php [hunk 9]
```diff
@@ -31,16 +31,18 @@ final class SessionConfiguratorPass implements CompilerPassInterface
      */
     public function process(ContainerBuilder $container)
     {
-        if (!$container->hasDefinition('session')) {
+        // @phpstan-ignore-next-line
```
Comment (fashxp) [inline bundles/CoreBundle/DependencyInjection/Compiler/SessionConfiguratorPass.php:34]: what is the reason for phpstan not happy here?

File: bundles/EcommerceFrameworkBundle/CartManager/SessionCart.php (hunk 4 above)
Comment (fashxp) [reply bundles/EcommerceFrameworkBundle/CartManager/SessionCart.php]: but for Pimcore 10 to be sure we don't add a BC break it would need a fallback to getting the session directly from the container as we did previously (and issue a deprecation warning). 
wdyt?

File: bundles/EcommerceFrameworkBundle/SessionEnvironment.php (hunk 6 above)
Comment (fashxp) [reply bundles/EcommerceFrameworkBundle/SessionEnvironment.php:76]: verified that request stack is always there ... but it potentially doesn't have a session set.

File: bundles/CoreBundle/Resources/config/services.yaml (hunk 3 above)
Comment (dvesh3) [reply bundles/CoreBundle/Resources/config/services.yaml:9]: Alias is still there from Symfony:
image

**Review by fashxp — CHANGES_REQUESTED** [review]

File: bundles/CoreBundle/DependencyInjection/Compiler/SessionConfiguratorPass.php (hunk 9 above)
Comment (dvesh3) [reply bundles/CoreBundle/DependencyInjection/Compiler/SessionConfiguratorPass.php:34]: The definitions are always there...
image

File: bundles/EcommerceFrameworkBundle/CartManager/SessionCart.php (hunk 4 above)
Comment (dvesh3) [reply bundles/EcommerceFrameworkBundle/CartManager/SessionCart.php]: done

File: bundles/EcommerceFrameworkBundle/PricingManager/PricingManager.php (hunk 5 above)
Comment (dvesh3) [reply bundles/EcommerceFrameworkBundle/PricingManager/PricingManager.php]: done

File: bundles/CoreBundle/DependencyInjection/Compiler/SessionConfiguratorPass.php (hunk 9 above)
Comment (fashxp) [reply bundles/CoreBundle/DependencyInjection/Compiler/SessionConfiguratorPass.php:34]: also in all supported symfony versions? in what environment is phpstan not happy there? on all versions? or only on specific?

Comment (dvesh3) [reply bundles/CoreBundle/DependencyInjection/Compiler/SessionConfiguratorPass.php:34]: Basically in highest dependencies test where the alias is there https://github.com/symfony/symfony/pull/38616/files#diff-830ba1cf3ce1e4b07ad2db84096e6b595b2f4cef60f4cdd5f2f7bc4608e7b287R47

File: bundles/EcommerceFrameworkBundle/PricingManager/PricingManager.php [hunk 10]
```diff
@@ -227,8 +230,9 @@ public function getValidRules()
...
         /** @var AttributeBagInterface $sessionBag */
-        $sessionBag = $this->session->getBag(SessionConfigurator::ATTRIBUTE_BAG_PRICING_ENVIRONMENT);
+        $sessionBag = $requestStack->getSession()->getBag(SessionConfigurator::ATTRIBUTE_BAG_PRICING_ENVIRONMENT);
```
Comment (fashxp) [inline bundles/EcommerceFrameworkBundle/PricingManager/PricingManager.php]: revert these changes and to back to the old `getEnvironment` implementation ... as there is no benefit yet. 

Add note to upgrade notes that we cannot get rid of all session injection deprecations due to BC reasons.

**Review by fashxp — CHANGES_REQUESTED** [review]
double check logs, if we can get rid of following message .... which pops up in demo 

```
[2022-06-30T11:36:21.423546+00:00] php.INFO: User Deprecated: Since symfony/framework-bundle 5.3: The "session.storage.factory.service" service is deprecated, use "session.storage.factory.native", "session.storage.factory.php_bridge" or "session.storage.factory.mock_file" instead. {"exception":"[object] (ErrorException(code: 0): User Deprecated: Since symfony/framework-bundle 5.3: The \"session.storage.factory.service\" service is deprecated, use \"session.storage.factory.native\", \"session.storage.factory.php_bridge\" or \"session.storage.factory.mock_file\" instead. at /var/www/html/var/cache/dev/Container1lLvexV/getSession_Storage_Factory_ServiceService.php:22)"} []
```

File: bundles/EcommerceFrameworkBundle/PricingManager/PricingManager.php (hunk 10 above)
Comment (dvesh3) [reply bundles/EcommerceFrameworkBundle/PricingManager/PricingManager.php]: Reverted https://github.com/pimcore/pimcore/pull/12241/commits/11e5a76953db7ab46e32f9050bf458157532de81

**Comment by dvesh3** [comment]
> double check logs, if we can get rid of following message .... which pops up in demo
> 
> ```
> [2022-06-30T11:36:21.423546+00:00] php.INFO: User Deprecated: Since symfony/framework-bundle 5.3: The "session.storage.factory.service" service is deprecated, use "session.storage.factory.native", "session.storage.factory.php_bridge" or "session.storage.factory.mock_file" instead. {"exception":"[object] (ErrorException(code: 0): User Deprecated: Since symfony/framework-bundle 5.3: The \"session.storage.factory.service\" service is deprecated, use \"session.storage.factory.native\", \"session.storage.factory.php_bridge\" or \"session.storage.factory.mock_file\" instead. at /var/www/html/var/cache/dev/Container1lLvexV/getSession_Storage_Factory_ServiceService.php:22)"} []
> ```

should be fixed by https://github.com/pimcore/pimcore/pull/12241/commits/eb19fac335d18040379f973a659f60bc0d8969e1

**Review by fashxp — APPROVED** [review]
